types/clownface/clownface-tests.ts

```diff
@@ -221,6 +221,19 @@ function testFilter() {
         const copy: never = quad;
         return true;
     });
+
+    const anyPointer: clownface.AnyPointer<clownface.AnyContext, Dataset> = <any> {};
+    function onlyVariables(ptr: clownface.GraphPointer): ptr is clownface.GraphPointer<Variable> {
+        return true;
+    }
+
+    const typeGuarded: clownface.AnyPointer<Variable, Dataset> = anyPointer.filter(onlyVariables);
+
+    function onlyNamedOrBlank(ptr: clownface.GraphPointer): ptr is clownface.GraphPointer<NamedNode> | clownface.GraphPointer<BlankNode> {
+        return true;
+    }
+
+    const multipleTypeGuarded: clownface.AnyPointer<NamedNode | BlankNode, Dataset> = anyPointer.filter<NamedNode | BlankNode>(onlyNamedOrBlank);
```

Comment:
How would I improve the type of `filter` so that the explicit type argument is not necessary?

```suggestion
    const multipleTypeGuarded: clownface.AnyPointer<NamedNode | BlankNode, Dataset> = anyPointer.filter(onlyNamedOrBlank);
```

Preferably also by allowing the type guard as `ptr is clownface.GraphPointer<NamedNode | BlankNode>`

Reply:
I'd recommend asking in the TypeScript Community Discord